nifpga/nifpga.py

```diff
@@ -369,90 +371,90 @@ def __init__(self):
-                    named_argtypes=[
-                        NamedArgtype("session", _SessionType),
-                        NamedArgtype("fifo", ctypes.c_uint32),
-                        NamedArgtype("data", ctypes.POINTER(type_ctype)),
-                        NamedArgtype("number of elements", ctypes.c_size_t),
-                        NamedArgtype("timeout ms", ctypes.c_uint32),
-                        NamedArgtype("elements remaining", ctypes.POINTER(ctypes.c_size_t)),
-                    ]),
-                LibraryFunctionInfo(
-                    pretty_name="WriteFifo%s" % datatype,
-                    name_in_library="NiFpgaDll_WriteFifo%s" % datatype,
-                    named_argtypes=[
-                        NamedArgtype("session", _SessionType),
-                        NamedArgtype("fifo", ctypes.c_uint32),
-                        NamedArgtype("data", ctypes.POINTER(type_ctype)),
-                        NamedArgtype("number of elements", ctypes.c_size_t),
-                        NamedArgtype("timeout ms", ctypes.c_uint32),
-                        NamedArgtype("empty elements remaining", ctypes.POINTER(ctypes.c_size_t)),
-                    ]),
-                LibraryFunctionInfo(
-                    pretty_name="AcquireFifoReadElements%s" % datatype,
-                    name_in_library="NiFpgaDll_AcquireFifoReadElements%s" % datatype,
-                    named_argtypes=[
-                        NamedArgtype("session", _SessionType),
-                        NamedArgtype("fifo", ctypes.c_uint32),
-                        NamedArgtype("elements", ctypes.POINTER(ctypes.POINTER(type_ctype))),
-                        NamedArgtype("elements requested ", ctypes.c_size_t),
-                        NamedArgtype("timeout ms", ctypes.c_uint32),
-                        NamedArgtype("elements acquired", ctypes.POINTER(ctypes.c_size_t)),
-                        NamedArgtype("elements remaining", ctypes.POINTER(ctypes.c_size_t)),
-                    ]),
-                LibraryFunctionInfo(
-                    pretty_name="AcquireFifoWriteElements%s" % datatype,
-                    name_in_library="NiFpgaDll_AcquireFifoWriteElements%s" % datatype,
-                    named_argtypes=[
-                        NamedArgtype("session", _SessionType),
-                        NamedArgtype("fifo", ctypes.c_uint32),
-                        NamedArgtype("elements", ctypes.POINTER(ctypes.POINTER(type_ctype))),
-                        NamedArgtype("elements requested ", ctypes.c_size_t),
-                        NamedArgtype("timeout ms", ctypes.c_uint32),
-                        NamedArgtype("elements acquired", ctypes.POINTER(ctypes.c_size_t)),
-                        NamedArgtype("elements remaining", ctypes.POINTER(ctypes.c_size_t)),
-                    ]),
-            ])  # end of library_function_infos.extend() call
-
+            if (datatype != DataType.FXP):
```

Comment:
It's hard to keep up with the indenting when the top of this giant "if" scrolls off the top of the screen. Can you just modify the "for" loop to iterate over a subset of DataType and then remove the "if" altogether?

Reply:
Update in "bitfile changes" I really wish we had a c# linq expression for this. since DataType is an Enum there is no great way to get a subset of it. I could great a list of the members of Datatype for a given condition, but we lose the methods the class has :( (specifically _return_ctype()). Hopefully I made it a bit by reversing the conditional so the tab does not go far down.